src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2674,11 +2656,11 @@ private void onApplicationDataInit(
                     final AmqpDataExFW dataEx = extension.get(amqpDataExRO::tryWrap);
                     assert dataEx != null;
 
+                    deliveryId++;
```

Comment:
Suggest moving this just above the call to `onApplicationDataInit` in `onApplicationData`.